solidity/contracts/converter/BancorConverter.sol

```diff
@@ -368,23 +410,24 @@ contract BancorConverter is IBancorConverter, SmartTokenController, ContractRegi
         require(_fromToken != _toToken); // validate input
 
         if (_toToken == token)
-            return getPurchaseReturn(_fromToken, _amount);
+            return getPurchaseReturn(_fromToken, _amount, 0);
         else if (_fromToken == token)
             return getSaleReturn(_toToken, _amount);
         else
-            return getCrossReserveReturn(_fromToken, _toToken, _amount);
+            return getCrossReserveReturn(_fromToken, _toToken, _amount, 0);
     }
 
     /**
       * @dev calculates the expected return of buying with a given amount of tokens
       * 
       * @param _reserveToken    contract address of the reserve token
       * @param _depositAmount   amount of reserve-tokens received from the user
+      * @param _delta           ???
       * 
       * @return amount of supply-tokens that the user will receive
       * @return amount of supply-tokens that the user will pay as fee
     */
-    function getPurchaseReturn(IERC20Token _reserveToken, uint256 _depositAmount)
+    function getPurchaseReturn(IERC20Token _reserveToken, uint256 _depositAmount, uint256 _delta)

```

Comment:
make it private, users can just call getReturn, simpler.
same for getSaleReturn and getCrossReserveReturn

Reply:
@yudilevi : I did. But then I found out that it breaks some of our tests. And since I consider these tests crucial for the current task at hand, I decided to leave those functions `public` for now.
Do you want me to make them `private`, and then add `public` functions which call the `private` ones with `delta = 0`?
I suggest we just leave it as is and take care of it later, after we know for sure that everything is working correctly.

Reply:
no, I want to make the private and remove the tests, or change the tests so that the will call these functions indirectly

Reply:
@yudilevi : Since we're planning to remove the `delta` solution, perhaps we should keep these tests?

Reply:
@yudilevi : Applied in 0.5.19 (since this is not directly related to removing ether-token), and then pulled to this branch.

Reply:
yeah I'd rather limit the API to getReturn - gives more flexibility and less dependency on external integrations